numpy/core/tests/test_mem_overlap.py

```diff
@@ -53,18 +51,7 @@ def _indices(ndims):
     """Returns ((axis0_src, axis0_dst), (axis1_src, axis1_dst), ... ) index pairs."""
 
     ind = _indices_for_axis()
-
-    # no itertools.product available in Py2.4
-
-    res = [[]]
-    for i in range(ndims):
-        newres = []
-        for elem in ind:
-            for others in res:
-                newres.append([elem] + others)
-        res = newres
-
-    return res
+    return itertools.product(ind, repeat=ndims)
```

Comment:
technically this returns a list of tuples instead of a list of lists but the only consumer is test_overlapping_assignments which doesn't care.